### PR TITLE
feat: add 'proper' argument parsing

### LINKED ISSUE #26: add proper GaussianSplattingViewer argument parsing
https://github.com/mosure/bevy_gaussian_splatting/blob/cd9fd074b95333dd57e3812b53376af378a0399a/viewer/viewer.rs#L61

```rust

        ..default()
    };

    // TODO: add proper GaussianSplattingViewer argument parsing
    let file_arg = std::env::args().nth(1);
    if let Some(n) = file_arg.clone().and_then(|s| s.parse::<usize>().ok()) {
        println!("generating {} gaussians", n);
        cloud = gaussian_assets.add(random_gaussians(n));

```

### PR DESCRIPTION
**Pull request by cs50victor**
use clap crate and enum for arg parsing.

- parsing to `usize` and then `usize` to back to `String` seems unnecessary / redundant. 

Closes #26

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -1,11 +1,35 @@
...
-use wasm_bindgen::prelude::*;
 #[cfg(target_arch = "wasm32")]
 use std::collections::HashMap;
+#[cfg(target_arch = "wasm32")]
+use wasm_bindgen::prelude::*;
 
+#[cfg(not(target_arch = "wasm32"))]
+#[derive(Debug, Parser)]
+#[command(about = "bevy gaussian splatting render pipeline plugin", version, long_about = None)]
+struct CliArgs {
```
Comment (mosure) [inline src/utils.rs:13]: I think merging `CliArgs` and `GaussianSplattingViewer` structs would be great (can happen in a future PR though, let me know your bandwidth): https://github.com/mosure/bevy_gaussian_splatting/blob/0c28bac7f071c729282a400db3a6701b44ef3590/viewer/viewer.rs#L47

e.g. this allows args like `--height`, `--width`, `--hide_fps`, `--hide_editor`, etc.

File: src/utils.rs [hunk 2]
```diff
@@ -15,14 +39,18 @@ pub fn setup_hooks() {
     }
 }
 
-
 #[cfg(not(target_arch = "wasm32"))]
-pub fn get_arg(n: usize) -> Option<String> {
-    std::env::args().nth(n)
+pub fn get_arg(arg: MainArgs) -> Option<String> {
```
Comment (mosure) [inline src/utils.rs:43]: converting this function to:
`pub fn get_args() -> CliArgs` (or `pub fn get_args() -> GaussianSplattingViewer` with the above suggestion) would simplify argument retrieval and avoid duplicate parse calls.

File: src/utils.rs (hunk 1 above)
Comment (cs50victor) [reply src/utils.rs:13]: makes sense

Comment (cs50victor) [reply src/utils.rs:13]: as for bandwidth, I plan on working / contributing on this project long term. I'll play around with merging the structs and see how it goes.

**Comment by mosure** [comment]
here is a good reference: https://github.com/mosure/bevy_args

**Pull request closed without merging**